### ISSUE TITLE
Stack overflow during unzip

### ISSUE THREAD

**Issue by aparshin** [EvanOxfeld/node-unzip#14]
I get stack overflow in the following example:

``` javascript
var unzip = require('unzip'),
    fs = require('fs');

unzip.Parse().end(fs.readFileSync('test.zip'));
```

Error:

```
node_modules/unzip/node_modules/over/overload.js:18
    var args = Array.prototype.slice.call(arguments);
                                     ^
RangeError: Maximum call stack size exceeded
```

Looks like the problem is in `inflateServiceRequest()` function (after commit 64199f4b6e4c8b9943b59cdf19378ccc02051f58).

zip is here: https://docs.google.com/file/d/0BwYD4dygLavbY3VBblM5QTBZVUE/edit?usp=sharing

**Comment by EvanOxfeld** [comment]
Thanks for reporting the issue. It should hopefully get fixed soon - I'm in the middle of extracting out much of inflateServiceRequest() into a faster and improved abstraction described in this [gist](http://bit.ly/Wk9xhn).

**Comment by EvanOxfeld** [comment]
unzip isn't using the compressed size for the second file within the zip. Fix forthcoming.

**Comment by EvanOxfeld** [comment]
Fix in 0.1.4 (commit 5653213)
